openai/client.go

```diff
@@ -160,7 +165,7 @@ func ConvertToMatrix(response *CreateEmbeddingResponse) [][]float32 {
 	return matrix
 }
 
-func (e *OpenAIEmbeddingFunction) CreateEmbedding(documents []string) ([][]float32, error) {
+func (e *OpenAIEmbeddingFunction) EmbedDocuments(ctx context.Context, documents []string) ([][]float32, error) {
 	response, err := e.apiClient.CreateEmbedding(&CreateEmbeddingRequest{
```

Comment:
let's also pass the context to CreateEmbedding and then pass it to the HTTP request